zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/distribution/DbDistributionState.java

```diff
@@ -109,6 +113,32 @@ public DbDistributionState(
             transactionContext,
             continuationByQueueKey,
             new PersistedCommandDistribution());
+
+    metrics = new DistributionMetrics(zeebeDb.getMeterRegistry());
+  }
+
+  @Override
+  public void onRecovered(final ReadonlyStreamProcessorContext context) {
```

Comment:
this one was quite tricky and I am not 100% certain if it's great. 🤯 

also the counters are all incremented during replay (they might also already been incremented before), thus we might count the events twice 🙃

Reply:
It only resets the (stateful) gauge values as they represent the db state, but doesn't touch the counters for example. If during replay an applier increments a counter, then they might be incremented multiple times per event (in case they were processed before and replayed again or multiple times due to restart loop or so). It also depends on the scraping cycle of prometheus and if during restart we somewhat clean up / reset the meter registry.

Reply:
Can we add a noop `DistributionMetrics` during replay and then swap with a real one when we start processing?

Reply:
You tell me 😄 , but that helps only with the counters incremented multiple times because of replay. Still if I understand correctly, it maybe also happen that the events haven't been counted before 🤔 thus, I believe we can't get something super consistent here. 

I believe in this thread it's more about the gauge recovery than the counters (were we accept, that they might be slightly off sometimes).

Reply:
That's what is happening here, right? Or do I misunderstand you point?

This transfers the count from rocksdb to the gauge meter
```
commandDistributionRecordColumnFamily.forEach(
        (distributionKey, record) -> getMetrics().addDistribution(distributionKey.getValue()));
```

Reply:
yeah, sorry  :see_no_evil: